### DISCUSSION TITLE
How can I save my query state value when I want to render to other page?

### DISCUSSION THREAD

**Discussion by virus231** [47ng/nuqs#474] [Q&A] [answered]

**Comment by franky47** [comment]
If you are running this client-side, and want to keep **all** of the search params, you could do this:

```ts
router.replace('/trip-list' + location.search)
```

Now to keep only a subset of those, based on their state value, there isn't currently an easy way to do that, but I just had this use-case too in a playground demo, so it's on the [roadmap](https://github.com/47ng/nuqs/issues/375).

What TanStack Router does to make routes with supported search params type-safe is insane for DX, hopefully we can get somewhere close with `nuqs` and Next.js typed routes in the future.

**Reply by virus231** [reply]
@franky47 
So, it's only one way for my issue, right?

`router.replace('/trip-list' + location.search)`

**Reply by franky47** [reply]
I've opened a PR for a potential solution for this, let me know if it would work for you: https://github.com/47ng/nuqs/pull/475

The approach of declaring an object where the keys are the search param names and the values are the parsers/serializers for each key opens up a lot of type-safe possibilities (when paired with `useQueryStates`, the server cache, and now this).

**Reply by virus231** [reply]
I think yes, when could I try this a potential solution?

**Reply by franky47** [reply]
I'll test-drive and polish the API over the weekend and push a beta version, watch the PR for the release bot announcing when it's available.

**Reply by franky47** [reply, marked as answer]
`1.16.0-beta.1` has been released with the `createSerializer` API, let me know how it works for you.

Documentation: https://github.com/47ng/nuqs#serializer-helper
